### PR TITLE
ci: Add action to generate stackaid.json

### PR DESCRIPTION
**Pull request by wescarr**
This action will generate a stackaid.json file, listing the dependencies that Sentry wants to fund on StackAid. FYI, I'm one of the @stackaid founders helping @chadwhitacre setup Sentry's funding.

### CODE REVIEW

File: .github/workflows/stackaid.yml [hunk 1]
```diff
@@ -0,0 +1,11 @@
...
+on:
+  push:
+    branches:
+      - main
+jobs:
+  stackaid-json:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - uses: stackaid/generate-stackaid-json@v1.9
```
Comment (vaind) [inline .github/workflows/stackaid.yml:11]: We pin actions in this repo. And checkout@v2 uses outdated Node, triggering notices.
```suggestion
      - uses: actions/checkout@v3
      - uses: stackaid/generate-stackaid-json@bbccf9adf6ab4acd69aa979b0cdc7e93a5e83d78 # pin@v1.9
```

File: .github/workflows/stackaid.yml [hunk 2]
```diff
@@ -0,0 +1,11 @@
+name: 'fund-on-stackaid'
+on:
+  push:
+    branches:
+      - main
```
Comment (vaind) [inline .github/workflows/stackaid.yml:5]: maybe update temporarily to trigger on this branch so we can see & check the recognized dependencies here?

**Comment by chadwhitacre** [comment]
Closing. See https://github.com/getsentry/sentry/pull/40553#issuecomment-1292120935. 🙏

File: .github/workflows/stackaid.yml (hunk 1 above)
Comment (wescarr) [reply .github/workflows/stackaid.yml:11]: Thanks for the feedback @vaind We'll update our README as well.

**Pull request closed without merging**